runtime/protocol/http/common/src/aws/smithy/kotlin/runtime/http/HttpBody.kt

```diff
@@ -191,6 +193,49 @@ public fun HttpBody.toHashingBody(
     else -> throw ClientException("HttpBody type is not supported")
 }
 
+/**
+ * Convert an [HttpBody] with an underlying [HashingSource] or [HashingByteReadChannel]
+ * to a [CompletingSource] or [CompletingByteReadChannel], respectively.
+ */
+@InternalApi
+public fun HttpBody.toCompletingBody(deferred: CompletableDeferred<String>): HttpBody = when (this) {
```

Comment:
These used to be `private`, now they are `@InternalApi public`, meaning they are included in our API dumps and part of backwards compatibility. Was this move necessary?

Reply:
Oops, I moved this here because I was going to add support for streaming checksums in the `HttpCheksumsRequired` interceptor